docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+  same behaviour as legacy sweep). If a transaction was committed, but **after** the *sweep timestamp*, we must not progress
+  targeted sweep past its start timestamp, so we remove it and all the writes with greater start timestamps from the batch.
+  4. Delete all writes that are referenced to from aborted transactions.
+  5. Partition the remaining `WriteInfo`s by `Cell` and `TableReference`, and then take the greatest start timestamp from each
+    partition. We only need one ranged tombstone per partition here, as all the other writes to that cell and table in this batch
+    are going to have a lower start timestamp and are therefore going to be deleted as well.
+       - If the strategy is conservative: write a garbage collection sentinel, then put a ranged tombstone deleting all versions of that
+         cell between 0 and write timestamp - 1, not deleting the sentinel or the write.
+       - If the strategy is thorough
+            - If the write was a tombstone, then put a ranged tombstone deleting all versions of that cell between -1 and write timestamp,
+              deleting both a potentially existing sentinel and the write.
+            - Otherwise, put a ranged tombstone deleting all versions of that cell between -1 and write timestamp - 1, deleting a
+              potentially existing sentinel, but not the write.
+  6. If the new sweep progress described in greater detail below guarantees that the minimum start timestamp swept in the future
+     will be in a greater fine, or coarse, partition than the previous *last swept timestamp*, then clean the sweep queue accordingly
+     as previously explained.
+  7. Persist the sweep progress in *sweepProgressPerShard* for the shard and strategy.
+  8. Finally, regardless of the success or failure of the iteration, unlock the TimeLock lock for the shard and strategy and
+     schedule the next iteration of sweep for the thread after a delay of 5 seconds.
+
+**Calculating Sweep Progress**:
+
+We wish to update progress to the greatest value we can guarantee we swept to. There are multiple cases to consider, in order:
+- If we do not find a candidate row of *sweepableCells* while reading from the sweep queue, we can update to *sweep timestamp - 1*.
+- If none of the timestamps from the batch were committed after the *sweep timestamp* and we have read all entries in
+  *sweepableCells* up to *sweep timestamp*, then we can update to *ts - 1*, where *ts* is the minimum between
+  the *sweep timestamp* and the first timestamp that would be written into the next row of *sweepableCells*.
+- Otherwise, update to *ts*, where *ts* is the greatest timestamp among the writes in the batch.
+
+## Consequences
+
+### Benefits
+- Targeted sweep can perform sweeps several orders of magnitude times faster than legacy sweep.
+- The load on the underlying KVS is significantly reduced, particularly in the case of Cassandra.
```

Comment:
^These 2 while likely to be true seem a bit premature for my taste, especially in non-Cassandra environments.